test/snippet/io/alignment_file/alignment_file_input_begin_and_front.cpp

```diff
@@ -0,0 +1,56 @@
+#include <fstream>
+#include <string>
+
+#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/io/alignment_file/input.hpp>
+#include <seqan3/io/stream/iterator.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/view/single_pass_input.hpp>
+#include <seqan3/range/view/take_until.hpp>
+#include <seqan3/std/algorithm>
+#include <seqan3/std/ranges>
+
+struct write_file_dummy_struct
+{
+    write_file_dummy_struct()
+    {
+
+auto sam_file_raw = R"(
+@HD VN:1.6 SO:coordinate
+@SQ SN:ref LN:45
+r001   99 ref  7 30 8M2I4M1D3M = 37  39 TTAGATAAAGGATACTG *
+r003    0 ref  9 30 5S6M       *  0   0 GCCTAAGCTAA       * SA:Z:ref,29,-,6H5M,17,0;
+r003 2064 ref 29 17 6H5M       *  0   0 TAGGC             * SA:Z:ref,9,+,5S6M,30,1;
+r001  147 ref 37 30 9M         =  7 -39 CAGCGGCAT         * NM:i:1
+)";
+
+        std::ofstream file_stream{"/tmp/my.sam"};
+        seqan3::ostreambuf_iterator stream_it{file_stream};
+
+        std::string sam_file{sam_file_raw};
+        auto sam_view = sam_file | seqan3::view::single_pass_input;
+
+        ++std::ranges::begin(sam_view); // skip first new line from raw string
+        std::ranges::copy(sam_view | seqan3::view::take_until(seqan3::is_char<' '>), stream_it);
+
+        while (begin(sam_view) != end(sam_view))
+        {
+            stream_it = '\t';
+            seqan3::detail::consume(sam_view | seqan3::view::take_until(!seqan3::is_char<' '>));
+            std::ranges::copy(sam_view | seqan3::view::take_until(seqan3::is_char<' '>), stream_it);
+        }
+    }
+};
+
+write_file_dummy_struct go{}; // just to have a temporary file accessible such that the snippet does not fail
+
+int main()
+{
+    seqan3::alignment_file_input fin{"/tmp/my.sam"};
```

Comment:
ok, while thinking about it. Please remove the entire boiler plate code above and only load the file from a istringstream. Also note, that you can have tabs in a string literal.